filesystem.go

```diff
@@ -58,6 +65,29 @@ func New(fs billy.Filesystem, path string) SivaFS {
 	}
 }
 
+// NewFilesystem creates an entire filesystem using siva as the main backend,
+// but supplying unsupported functionality using as a temporal files backend
+// the main filesystem
```

Comment:
What if `fs` already has a temp support?, after all, it is full billy.Filesystem.